lib/ui/screens/dashboard.dart

```diff
@@ -0,0 +1,132 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:kjg_muf_app/constants/kjg_colors.dart';
+import 'package:kjg_muf_app/ui/screens/dashboard_webview_screen.dart';
+import 'package:kjg_muf_app/ui/widgets/news_carousel_widget.dart';
+import 'package:kjg_muf_app/ui/widgets/newsletter_subscribe_button.dart';
+import 'package:kjg_muf_app/viewmodels/dashboard.viewmodel.dart';
+import 'package:kjg_muf_app/viewmodels/main.viewmodel.dart';
+import 'package:provider/provider.dart';
+
+class Dashboard extends StatelessWidget {
+  const Dashboard({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ChangeNotifierProvider(
+      create: (_) => DashboardViewModel(),
+      child: Builder(
+        builder: (context) {
+          MainViewModel mainViewModel = Provider.of<MainViewModel>(context);
+          final dashboardViewModel = Provider.of<DashboardViewModel>(context);
+          return Stack(
+            children: [
+              Align(
+                alignment: Alignment.bottomRight,
+                child: FractionallySizedBox(
+                  heightFactor: 0.3,
+                  child: Image.asset(
+                    "assets/mausis/mercimausi.png",
+                    fit: BoxFit.fitHeight,
+                    opacity: const AlwaysStoppedAnimation(.3),
+                  ),
+                ),
+              ),
+              Padding(
+                padding: const EdgeInsets.all(8.0),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(
+                      AppLocalizations.of(context)!
+                          .greeting(mainViewModel.firstName ?? "DU"),
+                      style: const TextStyle(
+                        fontWeight: FontWeight.w700,
+                        fontSize: 24.0,
+                        color: KjGColors.kjgLightBlue,
+                      ),
+                    ),
+                    const SizedBox(
+                      height: 8.0,
+                    ),
+                    Center(
+                      child: switch (dashboardViewModel.newsLoadingState) {
```

Comment:
Eventuell dieses ganze switch case noch in ein anderes Widget oder Funktion auslagern? Das ist ja für news und activities genau dasselbe.
Ich fände es cool wenn der Error und loading state die gleiche Höhe wie die tatsächlichen Posts hätten, dann verrutscht da nichts (evtl. beim Laden einfach eine graue Card mit Loading Indicator?)

Reply:
Ich habe das jetzt mit so einem Shimmer Loading Effect gelöst. Find das sieht am coolsten aus